File: server/src/main/scala/producer/message_generator/event_time_generator/EventTimeGenerator.scala
package producer.message_generator.event_time_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import producer.message_generator.data_generators.date_time.DateTimeGenerator

case class EventTimeGenerator(
    generator: DateTimeGenerator
)

object EventTimeGenerator:
    def fromPb(v: pb.EventTimeGenerator): EventTimeGenerator =
        EventTimeGenerator(
            generator = DateTimeGenerator.fromPb(v.generator.get)
        )

    def toPb(v: EventTimeGenerator): pb.EventTimeGenerator =
        pb.EventTimeGenerator(
            generator = Some(DateTimeGenerator.toPb(v.generator))
        )

File: ui/components/ui/ProducerSession/producer-task/message-generator/event-time-generator/event-time-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';
import { DateTimeGenerator, dateTimeGeneratorFromPb, dateTimeGeneratorToPb } from '../data-generators/date-time/date-time-generator';

export type EventTimeGenerator = {
  type: 'event-time-generator',
  generator: DateTimeGenerator
};

export function eventTimeGeneratorFromPb(v: pb.EventTimeGenerator): EventTimeGenerator {
  return {
    type: 'event-time-generator',
    generator: dateTimeGeneratorFromPb(v.getGenerator()!)
  };
}

export function eventTimeGeneratorToPb(v: EventTimeGenerator): pb.EventTimeGenerator {
  const generatorPb = new pb.EventTimeGenerator();
  generatorPb.setGenerator(dateTimeGeneratorToPb(v.generator));

  return generatorPb;
}
